cloud/scope/powervs_cluster_test.go

```diff
@@ -1704,3 +1704,521 @@ func TestCreateServiceInstance(t *testing.T) {
+		}
+		mockVPC.EXPECT().GetVPC(gomock.Any()).Return(nil, nil, fmt.Errorf("GetVPC returns error"))
+		requeue, err := clusterScope.ReconcileVPC()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+
+	t.Run("When VPC ID is already set and GetVPC returns empty output", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient: mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{VPC: &infrav1beta2.ResourceReference{ID: ptr.To("VPCID")}},
+			},
+		}
+		mockVPC.EXPECT().GetVPC(gomock.Any()).Return(nil, nil, nil)
+		requeue, err := clusterScope.ReconcileVPC()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+
+	t.Run("When VPC ID is already set and GetVPC returns vpc in pending state", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		vpcOutput := &vpcv1.VPC{Name: ptr.To("VPCName"), ID: ptr.To("VPCID"), Status: ptr.To(string(infrav1beta2.VPCStatePending))}
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient: mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{VPC: &infrav1beta2.ResourceReference{ID: ptr.To("VPCID")}},
+			},
+		}
+		mockVPC.EXPECT().GetVPC(gomock.Any()).Return(vpcOutput, nil, nil)
+		requeue, err := clusterScope.ReconcileVPC()
+		g.Expect(err).To(BeNil())
+		g.Expect(requeue).To(BeTrue())
+	})
+
+	t.Run("When VPC ID is already set and is valid", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		vpcOutput := &vpcv1.VPC{Name: ptr.To("VPCName"), ID: ptr.To("VPCID")}
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient: mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{VPC: &infrav1beta2.ResourceReference{ID: ptr.To("VPCID")}}},
+		}
+		mockVPC.EXPECT().GetVPC(gomock.Any()).Return(vpcOutput, nil, nil)
+		requeue, err := clusterScope.ReconcileVPC()
+		g.Expect(err).To(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+}
+
+func TestPowerVSScopeCreateVPC(t *testing.T) {
+	var (
+		mockVPC  *mock.MockVpc
+		mockCtrl *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockVPC = mock.NewMockVpc(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When resourceGroupID is nil", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient:      mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+		}
+
+		vpcID, err := clusterScope.createVPC()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(vpcID).To(BeNil())
+	})
+	t.Run("When resourceGroupID is not nil and create VPC is successful", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("When resourceGroupID is set and create VPC is successful", func(t *testing.T) {
```
Please change wherever applicable.